agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4440,32 +4440,41 @@ beerocks::message::sWifiChannel slave_thread::channel_selection_select_channel()
             continue;
         }
         for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
-            auto channel         = hostap_params.supported_channels[i];
-            auto operating_class = wireless_utils::get_operating_class_by_channel(
-                channel.channel, static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth));
+            auto channel        = hostap_params.supported_channels[i];
+            auto bw             = static_cast<beerocks::eWiFiBandwidth>(channel.channel_bandwidth);
+            auto center_channel = wireless_utils::get_5g_center_channel(channel.channel, bw, true);
+            auto operating_class =
+                wireless_utils::get_operating_class_by_channel(channel.channel, center_channel, bw);
```

Comment:
This makes little sense to me... The conversion of channel+bw to operating class should be one-to-one IMO. So *either* you pass just channel.channel and do the get_center_channel inside the function, *or* the function assumes that the center_channel is used for the 80MHz channels.

Reply:
Forget this.

What we really should have is something like a wireless_operating_channel class which has constructors based on (operatingclass, channel) and on (channel, bandwidth). This class would be used everywhere internally, and it would have member functions to convert internally to whatever format is needed.

That was actually my plan all along when operating classes were introduced.

However, changing the code like that is *a lot* of work, with limited benefit. So let's just forget about it.

So for this PR, I'm OK with the proposal here. get_operating_class_by_channel *always* needs to be combined with the get_center_frequency because that's what is needed for the "operating class" form.

Or maybe one improvement would be to introduce just a single additional function in wireless_utils: `get_operating_class_channel(channel, bandwidth)`. So when converting to the "operating class" format, you derive both the operating class itself and the "operating class channel" i.e. the channel as used in the "operating class" format. Does that make sense?

Reply:
Why forget this? `get_operating_class_by_channel` should accpet a single parameter, so swifiChannel makes perfect sense, this until we have the wireless_operating_channel class which I like even more but definitely not for this PR.

Anyway, see if 6b4dd98 is better.